Using 'runtimes' in global.json fails on Linux/macOS - Error MSB4018: The "InstallDotNetCore" task failed unexpectedly

Trying to use a new feature of Arcade (https://github.com/dotnet/arcade/pull/2343) in Arcade v1.0.0-beta.19256.12.

My global.json file:
```json
{
  "sdk": {
    "version": "3.0.100-preview5-011568"
  },
  "tools": {
    "dotnet": "3.0.100-preview5-011568",
    "runtimes": {
      "dotnet": [
        "$(MicrosoftNETCoreAppPackageVersion)"
      ]
    }
  },
  "msbuild-sdks": {
    "Microsoft.DotNet.Arcade.Sdk": "1.0.0-beta.19256.12",
    "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.19256.12"
  }
}

```

On macOS/Linux, the build.sh script fails on the  InstallDotNetCore target with this error.

```
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018: The "InstallDotNetCore" task failed unexpectedly. [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018: System.ComponentModel.Win32Exception (13): Permission denied [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at System.Diagnostics.Process.Start() [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at System.Diagnostics.Process.Start(ProcessStartInfo startInfo) [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at Microsoft.DotNet.Arcade.Sdk.InstallDotNetCore.Execute() in /_/src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetCore.cs:line 102 [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/InstallDotNetCore.targets(12,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/vsts/work/1/s/.packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19256.12/tools/Tools.proj]
```

see https://github.com/aspnet/Extensions/pull/1692

cc @chcosta

--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/InstallDotNetCore.targets
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/InstallDotNetCore.targets
@@ -8,6 +8,9 @@
       <_DotNetInstallScript>$(RepositoryEngineeringDir)common\dotnet-install.cmd</_DotNetInstallScript>
       <_DotNetInstallScript Condition="'$(OS)' != 'Windows_NT'">$(RepositoryEngineeringDir)common\dotnet-install.sh</_DotNetInstallScript>
     </PropertyGroup>
+    
+    <Exec Condition="'$(OS)' != 'Windows_NT'"
+          Command="chmod +x &quot;$(_DotNetInstallScript)&quot;" />
 
     <InstallDotNetCore
       VersionsPropsPath="$(RepoRoot)eng\Versions.props"

